client/landing/stepper/declarative-flow/internals/steps-repository/automated-copy-site/index.tsx

```diff
@@ -0,0 +1,170 @@
+	);
+	const sourceSiteId = sourceSite?.ID;
+	const { requestLatestAtomicTransfer } = useDispatch( SITE_STORE );
+	const { getSiteLatestAtomicTransfer, getSiteLatestAtomicTransferError } = useSelect( ( select ) =>
+		select( SITE_STORE )
+	);
+
+	useEffect( () => {
+		if ( isCopying && site ) {
+			const timer = setInterval( async () => {
+				await requestLatestAtomicTransfer( site.ID );
+				const transfer = getSiteLatestAtomicTransfer( site.ID );
+				const transferError = getSiteLatestAtomicTransferError( site.ID );
+				const transferStatus = transfer?.status;
+				const isTransferringStatusFailed = transferError && transferError?.status >= 500;
+
+				if ( isTransferringStatusFailed ) {
+					setIsError( 'Error Copying' );
+				} else if ( transferStatus === 'completed' ) {
+					setIsSuccess();
+				}
+			}, TIME_CHECK_TRANSFER_STATUS );
+			return () => clearTimeout( timer );
+		}
+	}, [
+		getSiteLatestAtomicTransfer,
+		getSiteLatestAtomicTransferError,
+		isCopying,
+		requestLatestAtomicTransfer,
+		setIsError,
+		setIsSuccess,
+		site,
+	] );
+
+	const startCopying = useCallback(
+		function startCopying() {
+			if ( ! site?.ID || ! sourceSiteId ) {
+				return;
+			}
+			return wpcom.req
+				.post( {
+					path: `/sites/${ site.ID }/copy-from-site`,
+					apiNamespace: 'wpcom/v2',
+					body: {
+						source_blog_id: sourceSiteId,
+					},
+				} )
+				.then( () => {
+					setIsCopying();
+				} )
+				.catch( ( e: { message: string; code: string; data: { status: number } } ) => {
+					// eslint-disable-next-line no-console
+					console.error( e );
+					setIsError( e.message || 'Error Starting the Copy' );
+				} );
+		},
+		[ setIsCopying, setIsError, site?.ID, sourceSiteId ]
+	);
+
+	useEffect( () => {
+		startCopying();
+	}, [ startCopying ] );
+
+	if ( ! sourceSlug || ! siteSlug ) {
+		// TODO: show a better error or redirect to the start /sites
+		return <h1>{ __( 'Missing required parameters' ) }</h1>;
+	}
+
+	if ( ! sourceSiteId || ! site ) {
+		// eslint-disable-next-line no-console
+		console.info( { sourceSlug, siteSlug, sourceSiteId, site } );
+		// Wait for stores to populate
+		return null;
+	}
+
+	return (
+		<StepContainer
+			stepName="AutomatedcopySite"
+			hideBack
+			recordTracksEvent={ recordTracksEvent }
+			stepContent={
+				<div>
+					{ isLoading && <h1>{ __( 'Copying Site…' ) }</h1> }
+					{ isError && (
+						<>
+							<h1>{ __( 'Error copying site. Contact support' ) }</h1>
+							<p>{ error }</p>
+							<button onClick={ startCopying }>{ __( 'Retry?' ) }</button>
+						</>
+					) }
+					{ isSuccess && (
+						<>
+							<h1>{ __( 'Site copied successfully 🥳' ) }</h1>
+							<p>
+								{ __(
+									'Congrats all your content has been migrated and your new site is ready to enjoy.'
+								) }
+							</p>
+							<button onClick={ () => submit?.( { siteSlug } ) }>
+								{ __( 'Visit my new site' ) }
+							</button>
```

Comment:
Maybe we should simply redirect them to their site on success?

Reply:
Done by @kozer 🥳 
- https://github.com/Automattic/wp-calypso/pull/72024